src/components/general/input/index.tsx

```diff
@@ -0,0 +1,20 @@
+import styled from 'styled-components';
+
+const StyledInput = styled.input`
+  border-radius: 20px;
+  border: ${({ theme }) => theme.colors.dark} solid 1px;
+  margin: ${({ theme }) => theme.sizes.spacing.sm};
+  padding: ${({ theme }) => theme.sizes.spacing.sm}
+    ${({ theme }) => theme.sizes.spacing.lg};
+  font-size: ${({ theme }) => theme.sizes.fonts.lg};
+  font-weight: 300;
+`;
+
+interface InputProps extends React.ComponentPropsWithoutRef<'input'> {
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const Input: React.FC<InputProps> = (props) => {
+  const { ...rest } = props;
```

Comment:
Because you're not passing additional props, you can get rid of line 18 where you get the rest and just return <StyledInput {...props} />;

We want to use ...rest when we need to extract other props that our component might need to configure the input somehow. For example:

`
const {isButton, ...rest} = props

if (isButton) return <StyledInput type="button" {...rest} />;

return <StyledInput {...rest} />;
`

This is a silly example, but it's just to explain that we might get extra props to configure this. If we don't expect any, you don't need ...rest